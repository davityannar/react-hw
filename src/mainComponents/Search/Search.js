import React, { Component } from 'react'
import {connect} from 'react-redux'
import {asyncHandlingSubmitFormAction, changingSearchInputValue} from '../../actions/actionCreators'

import SearchInput from './SearchInput/SearchInput'
import SearchButton from './SearchButton/SearchButton'
import SearchFilters from './SearchFilters/SearchFilters'
import './Search.scss'

import { Link,withRouter  } from 'react-router-dom'

export class Search extends Component {
  constructor(props) {
    super(props);
  }

  submitForm(e) {
    e.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.submitForm.bind(this)}>
        <SearchInput placeholder='Example: Quentin Tarantino' inputClass='search' passingValue={this.props.searchValue}  searchValue={this.props.searchValue} onCustomChange={this.props.changingInputValue} />
        <div className='header-filters' >
          <SearchFilters />
          <Link to={{pathname: `/movies`, search: `?search=${this.props.searchValue}&searchBy=${this.props.searchBy}&sortBy=${this.props.sortBy}&sortOrder=desc&limit=20`}}><SearchButton buttonClass='filter' onCustomSubmit={() => this.props.onSubmitForm(this.props.searchValue, this.props.searchBy, this.props.sortBy)} >Search</SearchButton></Link>
        </div>
      </form>
    )
  }
}

const mapStateToProps = (state) => ({searchValue: state.search.searchValue, searchBy: state.search.searchBy, sortBy: state.search.sortBy})
const mapDispatchToProps = (dispatch) => ({
  changingInputValue(e) {
    dispatch(changingSearchInputValue(e.target.value))
  },
  onSubmitForm(searchText, searchBy, sortBy) {
    dispatch(asyncHandlingSubmitFormAction(searchText, searchBy, sortBy));
  }
})

export default  withRouter(connect(mapStateToProps,mapDispatchToProps)(Search))



