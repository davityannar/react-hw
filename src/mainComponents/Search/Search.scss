.header-filters {
    display: flex;
    justify-content: space-between;
    margin-bottom: 40px;
  }

  .inputClass {
    margin-bottom: 30px;
    color: #fff;
    background-color: #000;
    padding: 13px 20px;
    width: 100%;
    border: none;
    outline: none;
    border-bottom: 1px solid #f55263;
  
    &::placeholder {
      color: #fff;
    }
}


.search-filters {
  display: flex;
  color: #fff;

  p {
      align-self: center;
      margin: 0;
  }
  * {
      text-transform: uppercase;
  }
}

.filter {
  display: none;

& + label {
  background-color: #666;
  padding: 4px 15px;
  margin-left: 20px;
  border-radius: 2px;
  cursor: pointer;
}

&:checked + label {
  background-color: #f55263;
  padding: 4px 15px;
  margin-left: 20px;
  border-radius: 2px;
  cursor: pointer;
}
}

.filter1 {
  width: 110px;
  text-align: center;
  text-transform: uppercase;
  color: #fff;
  background-color: #f55263;
  outline: none;
  border: none;
  padding: 6px 0;
  border-radius: 3px;
  cursor: pointer;
}