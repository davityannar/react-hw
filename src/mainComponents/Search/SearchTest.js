import React from 'react'
import {connect} from 'react-redux'
import {buttonClick} from '../../actions/actionCreators'

import './SearchButton.scss'

const SearchButton = (props) => {
  let buttonClass = '';
  if (props.buttonClass === 'filter') {
    buttonClass = 'search-button';
  }

  return (
    <div>
      <button className={buttonClass}>{props.buttonText}</button>
      <button className={buttonClass} onClick={props.changingBtnText} >{props.btnText}</button>
    </div>
  )
}

export default SearchButton
const mapStateToProps = (state) => ({btnText: state.btnText})
const mapDispatchToProps = (dispatch) => ({
  changingBtnText() {
    dispatch(buttonClick('Ogo'))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(SearchButton)