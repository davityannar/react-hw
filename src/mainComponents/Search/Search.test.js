import React from 'react'
import Search from './Search'

describe('Tests Search component', () => {
  it('Matching snapshot', () => {
    const component = shallow(<Search />)

    expect(component).toMatchSnapshot();
  })
})