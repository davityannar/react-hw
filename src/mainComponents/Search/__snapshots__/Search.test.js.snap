// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Search component Matching snapshot 1`] = `
<form>
  <div>
    <input
      className="inputClass"
      placeholder="Quentin Tarantino"
    />
  </div>
  <div
    className="header-filters"
  >
    <div
      className="search-filters"
    >
      <p>
        Search by
      </p>
      <input
        className="filter"
        defaultChecked={true}
        id="title"
        name="search"
        type="radio"
        value="TITLE"
      />
      <label
        htmlFor="title"
      >
        Title
      </label>
      <input
        className="filter"
        id="genre"
        name="search"
        type="radio"
        value="GENRE"
      />
      <label
        htmlFor="genre"
      >
        Genre
      </label>
    </div>
    <div>
      <button
        className="filter1"
        onClick={[Function]}
      >
        search
      </button>
    </div>
  </div>
</form>
`;
